From 5b2886ff893cdaa15df6302c3a33ca5320a1883b Mon Sep 17 00:00:00 2001
From: Kate Montgomery <katemontgomery@google.com>
Date: Thu, 26 Jan 2023 18:31:45 +0000
Subject: [PATCH] Fix bypass BAL via LocationManager.requestFlush

Bug: 235823542
Test: atest LocationProviderManagerTest and manual tests
Change-Id: I2a0fa7b99c3ad5ae839d8018ec70cb5c26e33240
(cherry picked from commit 80bc46c48cb693ede724fac070a87df30d813efc)
Merged-In: I2a0fa7b99c3ad5ae839d8018ec70cb5c26e33240
---
 .../server/location/provider/LocationProviderManager.java        | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/location/provider/LocationProviderManager.java b/services/core/java/com/android/server/location/provider/LocationProviderManager.java
index 1235352b0590..f0aff2a503b0 100644
--- a/services/core/java/com/android/server/location/provider/LocationProviderManager.java
+++ b/services/core/java/com/android/server/location/provider/LocationProviderManager.java
@@ -300,6 +300,7 @@ public class LocationProviderManager extends
         public void deliverOnFlushComplete(int requestCode) throws PendingIntent.CanceledException {
             BroadcastOptions options = BroadcastOptions.makeBasic();
             options.setDontSendToRestrictedApps(true);
+            options.setPendingIntentBackgroundActivityLaunchAllowed(false);
 
             mPendingIntent.send(mContext, 0, new Intent().putExtra(KEY_FLUSH_COMPLETE, requestCode),
                     null, null, null, options.toBundle());
-- 
2.39.1.456.gfc5497dd1b-goog

